kstreams-live-update/aggregator/src/main/java/io/debezium/examples/kstreams/liveupdate/aggregator/StreamsPipelineManager.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.examples.kstreams.liveupdate.aggregator;
+
+import java.io.IOException;
+import java.util.Properties;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.KafkaStreams;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.StreamsConfig;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.Produced;
+import org.apache.kafka.streams.kstream.Windowed;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.examples.kstreams.liveupdate.aggregator.cdi.Eager;
+import io.debezium.examples.kstreams.liveupdate.aggregator.serdes.StringWindowedSerde;
+import io.debezium.examples.kstreams.liveupdate.aggregator.ws.ChangeEventsWebsocketEndpoint;
+
+/**
+ * Starts up the KStreams pipeline once the source topics have been created.
+ *
+ * @author Gunnar Morling
+ */
+@ApplicationScoped
+@Eager
+public class StreamsPipelineManager {
+
+    private static final Logger LOG = LoggerFactory.getLogger( StreamsPipelineManager.class );
+
+    @Inject
+    @ConfigProperty(name="kafka.bootstrap.servers", defaultValue="kafka:9092")
+    private String kafkaBootstrapServers;
+
+    @Inject
+    private ChangeEventsWebsocketEndpoint websocketsEndPoint;
+
+    private KafkaStreams streams;
+    private ExecutorService executor;
+
+    @PostConstruct
+    public void startKStreams() {
```

Comment:
Replace 
```
    @PostConstruct
    public void startKStreams() {
```
with `public void startKStreams(@Observes @Initialized(ApplicationScoped.class) Object init) {`

and there is no need for the eager stuff.

Reply:
Hum, so I have tried that, but without luck. The method isn't invoked, which makes sense as the application scoped bean will be loaded lazily. So I'm going to go with the original proposal, happy to adjust if there's an easier way to make it happen.